airflow/timetables/events.py

```diff
@@ -79,22 +82,34 @@ def next_dagrun_info(
         last_automated_data_interval: DataInterval | None,
         restriction: TimeRestriction,
     ) -> DagRunInfo | None:
-        if last_automated_data_interval is None:
-            next_event = self.event_dates[0]
-        else:
-            future_dates = itertools.dropwhile(
-                lambda when: when <= last_automated_data_interval.end,  # type: ignore
-                self.event_dates,
-            )
-            next_event = next(future_dates, None)  # type: ignore
-            if next_event is None:
-                return None
+        earliest = restriction.earliest
+        if not restriction.catchup:
+            current_time = pendulum.DateTime.utcnow()
```

Comment:
I would recommend to use `airflow.utils.timezone.utcnow` for this one instead of direct use `pendulum.DateTime.utcnow()`.

We might potentially get rid of `pendulum` in the future, and this might help to replace this easier